src/logging.cc

```diff
@@ -1215,10 +1214,14 @@ void LogMessage::Init(const char* file,
   allocated_ = NULL;
   if (severity != GLOG_FATAL || !exit_on_dfatal) {
 #ifdef GLOG_THREAD_LOCAL_STORAGE
+    const uintptr_t kAlign = sizeof(void*) - 1;
+
     // No need for locking, because this is thread local.
     if (thread_data_available) {
       thread_data_available = false;
-      data_ = new (&thread_msg_data) LogMessageData;
+      char* align_ptr = (char*)(uintptr_t(thread_msg_data + kAlign) & (~kAlign));
+      assert(uintptr_t(align_ptr) % sizeof(void*) == 0);
```

Comment:
Ideally, this should be checked at compile time.

Reply:
Expressions that are run-time in their nature can not be checked at compile time. This one depends on the value of `align_ptr`. Would you like me to remove this line?

Reply:
Are you sure about that? The `align_ptr` depends on `&threads_msg_data` and `kAlign` which are not runtime expressions. If there is no safer way around it, you can keep the `assert`.

Reply:
This statement is incorrect.
thread_msg_data can not be constexpr pointer because even if compiler knows the segment offset of the pointer, its process address will be determined at linking time. In addition it's thread local which means it's in fact allocated during the run-time.
https://godbolt.org/z/0uiba1 
I will bring back the assert.